frb_example/benches/rust/src/executor.rs

```diff
@@ -0,0 +1,237 @@
+        value: Some(elapsed),
+        extra: None,
+        unit,
+    });
+}
+
+/// simple newtype pattern
+///
+/// this is the bridge's custom [error handler](http://cjycode.com/flutter_rust_bridge/feature/handler.html#example-report-errors-to-your-backend-in-addition-to-telling-dart).
+#[derive(Clone, Copy)]
+pub struct BenchErrorHandler(ReportDartErrorHandler);
+
+impl Default for BenchErrorHandler {
+    fn default() -> Self {
+        Self(ReportDartErrorHandler)
+    }
+}
+
+/// trait bound required by [`SimpleHandler`]
+impl ErrorHandler for BenchErrorHandler {
+    fn handle_error(&self, port: MessagePort, error: handler::Error) {
+        self.0.handle_error(port, error)
+    }
+
+    fn handle_error_sync(
+        &self,
+        error: flutter_rust_bridge::handler::Error,
+    ) -> flutter_rust_bridge::support::WireSyncReturnStruct {
+        self.0.handle_error_sync(error)
+    }
+}
+
+/// a slightly more elaborated newtype
+///
+/// this is the bridge's custom [executor](http://cjycode.com/flutter_rust_bridge/feature/handler.html#example-log-when-execution-starts-and-ends).
+pub struct BenchExecutor {
+    inner: ThreadPoolExecutor<BenchErrorHandler>,
+    json: bool,
+}
+
+/// subscribe for tracing only on native platforms
+#[cfg(not(target_family = "wasm"))]
+fn maybe_subscribe(json: bool) {
+    use tracing_subscriber::FmtSubscriber;
+    let subscriber = if json {
+        None
+    } else {
+        Some(
+            FmtSubscriber::builder()
+                .with_max_level(tracing::Level::TRACE)
+                .finish(),
+        )
+    };
+    if let Some(subscriber) = subscriber {
+        tracing::subscriber::set_global_default(subscriber)
+            .expect("Setting default subscriber failed");
+    }
+}
+
+/// on web platform, this is a no-op
+#[cfg(target_family = "wasm")]
+#[inline(always)]
+fn maybe_subscribe(_: bool) {}
+
+impl BenchExecutor {
+    /// also retrieve env var to set mode to **stdout** or **json**
+    /// and subscribe to `tracing` if needed
+    pub(crate) fn new(error_handler: BenchErrorHandler) -> Self {
+        let json = std::env::var("JSON")
+            .unwrap_or_else(|_| "false".into())
+            .parse::<bool>()
+            .expect("Invalid JSON env var (expected boolean)");
+        maybe_subscribe(json);
+        Self {
+            inner: ThreadPoolExecutor::new(error_handler),
+            json,
+        }
+    }
+}
+
+impl Executor for BenchExecutor {
+    /// wrap [`bench_around`] over async FFI calls
+    fn execute<TaskFn, TaskRet>(&self, wrap_info: WrapInfo, task: TaskFn)
+    where
+        TaskFn: FnOnce(TaskCallback) -> anyhow::Result<TaskRet> + Send + UnwindSafe + 'static,
+        TaskRet: IntoDart,
+    {
+        let debug_name_string = wrap_info.debug_name.to_string();
+        let json = self.json;
+        self.inner.execute(wrap_info, move |task_callback| {
+            Self::bench_around(&debug_name_string, json, move || task(task_callback))
```

Comment:
Oops, seems like we are recording each single frb call... Indeed, to perform microbenchmarks, seem that we should  execute many times and record once. For example, below pseudo code:

```
// no
for(var i in range(10000)) {
  var start = now();
  api.call_frb_method();
  record(now() - start());
}

// yes
var start = now();
for(var i in range(10000)) {
  api.call_frb_method();
}
record((now() - start()) / 10000);
```

Reply:
yeah I know that but I don't intend to even try to implement something like e.g. criterion lol
but it can more or less be achieved out-of-the-box by mixing custom `Interceptor` and `benchmark_harness`.
`benchmark_harness` already provides the `warm-up` mechanism.

Reply:
when running `bench-simple` (as renamed)

https://user-images.githubusercontent.com/21016014/194591592-fc7412dd-eede-43dd-99ee-b009299ed249.mov

I think there's already all the building blocks to iterate on it :)

Reply:
Well seems they are quite different. IMHO for (micro) benchmarking we should not do measurement every time, but should only measure time every N times. Correct me if I am wrong!

Reply:
(See my comments below)